core/src/com/unciv/ui/newgamescreen/PlayerPickerTable.kt

```diff
@@ -71,12 +72,21 @@ class PlayerPickerTable(val previousScreen: IPreviousScreen, var gameParameters:
             playerListTable.add("+".toLabel(Color.BLACK, 30).apply { this.setAlignment(Align.center) }
                     .surroundWithCircle(50f).onClick {
                         var player = Player()
-                        if (noRandom) { player = Player(getAvailablePlayerCivs().first().name) }
+                        // no random mode - add first not spectator civ if still available
+                        if (noRandom) {
+                            try {
```

Comment:
What kind of error can you encounter here? Can you catch it with an if(...) instead of with a try/catch?
If so please do - better for debugging, performance, readability

Reply:
Agree, seems to be complicated even for me right now. Will try to refactor.
The main idea is when we already added all nations from getAvailablePlayerCivs().first will throw an exeption. So I need to handle that rare case.